apps/api/src/app/environments-v1/usecases/outputRendererUseCases.ts

```diff
@@ -0,0 +1,144 @@
+// Base interface for all renderers
+import { render } from '@maily-to/render';
+import {
+  ChatRenderOutput,
+  EmailRenderOutput,
+  InAppRenderOutput,
+  PushRenderOutput,
+  RedirectTargetEnum,
+  SmsRenderOutput,
+  TipTapNode,
+} from '@novu/shared';
+import { z } from 'zod';
+import { ExpendEmailEditorSchemaUseCase } from '../../environments/render/email-schema-extender';
+
+class RenderCommand {
+  controlValues: Record<string, unknown>;
+}
+
+export class EmailOutputRendererUseCase {
+  async execute(renderCommand: RenderCommand): Promise<EmailRenderOutput> {
+    const parse = EmailStepControlSchema.parse(renderCommand.controlValues);
+    const schema = parse.emailEditor as TipTapNode;
+    const expandedSchema = new ExpendEmailEditorSchemaUseCase().execute({ schema });
+    const html = await render(expandedSchema);
+
+    return { subject: parse.subject, body: html };
+  }
+}
+
+// Concrete Renderer for Chat Preview
+export class ChatOutputRendererUseCase {
+  execute(renderCommand: RenderCommand): ChatRenderOutput {
+    const body = (renderCommand.controlValues.body as string) || 'Default chat message';
```

Comment:
Should we maybe just default to ''?

Reply:
removed the default, the as is critical for compilation